diff -ruN a/Library/PackageCache/com.unity.ai.assistant@5ecf2493a698/Editor/UI/Scripts/Components/AssistantTextField.cs b/Packages/com.unity.ai.assistant/Editor/UI/Scripts/Components/AssistantTextField.cs
--- a/Library/PackageCache/com.unity.ai.assistant@5ecf2493a698/Editor/UI/Scripts/Components/AssistantTextField.cs	2026-02-24 06:58:07
+++ b/Packages/com.unity.ai.assistant/Editor/UI/Scripts/Components/AssistantTextField.cs	2026-02-24 16:47:54
@@ -76,6 +76,7 @@
         bool m_HighlightFocus;
         bool m_EditContextEnabled;
         bool m_ImageSizeExceeded;
+        string m_PreviousComposition;
 
         public AssistantTextField()
             : base(AssistantUIConstants.UIModulePath)
@@ -345,6 +346,7 @@
 
         void OnTextFieldValueChanged(ChangeEvent<string> evt)
         {
+            m_PreviousComposition = Input.compositionString;
             OnChatValueChanged();
         }
 
@@ -653,6 +655,8 @@
                 case KeyCode.Return:
                 case KeyCode.KeypadEnter:
                     if (Text.Trim().Length == 0)
+                        return;
+                    if (!string.IsNullOrEmpty(Input.compositionString) || !string.IsNullOrEmpty(m_PreviousComposition))
                         return;
                     break;
 
diff -ruN a/Library/PackageCache/com.unity.ai.assistant@5ecf2493a698/Editor/UI/Styles/AssistantSharedDark.uss b/Packages/com.unity.ai.assistant/Editor/UI/Styles/AssistantSharedDark.uss
--- a/Library/PackageCache/com.unity.ai.assistant@5ecf2493a698/Editor/UI/Styles/AssistantSharedDark.uss	2026-02-24 06:58:07
+++ b/Packages/com.unity.ai.assistant/Editor/UI/Styles/AssistantSharedDark.uss	2026-02-24 16:48:31
@@ -46,6 +46,10 @@
     --muse-chat-popup-tab-hover: #cbcbcb;
 }
 
+.unity-text-element {
+    -unity-font-definition: url("project://database/Packages/jp.keijiro.ai.assistant.extensions/Editor/NotoSansJP-Regular.ttf?fileID=12800000&guid=f7d2dd1a328194df9b71491ff2d07e2b&type=3#NotoSansJP-Regular");
+}
+
 .-muse-chat-panel {
     background-color: var(--muse-chat-backgrounds-150);
 }
