--- Library/PackageCache/com.unity.ai.assistant@f231d0317302/Editor/UI/Scripts/Components/AssistantTextField.cs	2026-01-13 01:42:34
+++ Packages/com.unity.ai.assistant/Editor/UI/Scripts/Components/AssistantTextField.cs	2026-01-14 16:52:48
@@ -56,6 +56,7 @@
         bool m_HighlightFocus;
         bool m_EditContextEnabled;
         bool m_ImageSizeExceeded;
+        string m_PreviousComposition;
 
         public AssistantTextField()
             : base(AssistantUIConstants.UIModulePath)
@@ -229,6 +230,7 @@
 
         void OnTextFieldValueChanged(ChangeEvent<string> evt)
         {
+            m_PreviousComposition = Input.compositionString;
             OnChatValueChanged();
         }
 
@@ -379,6 +381,8 @@
                 case KeyCode.KeypadEnter:
                     if (Text.Trim().Length == 0)
                         return;
+                    if (!string.IsNullOrEmpty(Input.compositionString) || !string.IsNullOrEmpty(m_PreviousComposition))
+                        return;
                     break;
 
                 default:
